app/api/dds/v1/search_api.rb

```diff
@@ -76,12 +78,81 @@ class SearchAPI < Grape::API
           raise NameError.new("object_kind #{included_kind} Not Supported") unless kinded_model
           indices << kinded_model
         end
-        ElasticsearchResponse.new(
+        DeprecatedElasticsearchResponse.new(
           query: search_params[:search_query],
           indices: indices,
           policy_scope: method(:policy_scope)
         )
       end
+
+      desc 'Search Folders and Files' do
+        detail 'This endpoint allows searches of folders and files with a variety of query, filter, and post_filter features'
+        named 'search folders_files'
+        failure [
+          [200, 'This will never happen'],
+          [201, 'Success'],
+          [401, 'Unauthorized'],
+          [400, 'Invalid value specified for query_string.fields'],
+          [400, 'Invalid "{key: value}" specified for filters[].object'],
+          [400, 'Required fields - aggs[].field, aggs[].name must be specified'],
+          [400, 'Invalid value specified for aggs[].field'],
+          [400, 'Invalid size specified for aggs[].size - out of range'],
+          [400, 'Invalid "{key: value}" specified for post_filters[].object']
+        ]
+      end
+      params do
+        optional :query_string, type: Hash do
+          optional :query, type: String, desc: "The string fragment (phrase) to search for in query_string.fields. Required if query_string.fields are submitted."
+          optional :fields, type: Array, desc: "List of text fields to search; valid options are name and tags.label.  If not specified, defaults to all options. Requires query_string.query."
+        end
+        optional :filters, type: Array, desc: 'List of boolean search objects (predicates) that are joined together by AND operator, represented as key: value pairs' do
+          optional :kind , type: Array, desc: 'Limit search context to list of kinds; valid options are dds-folder and dds-file - requires a set of comma-delimited kinds like so: {"kind": ["dds-file"]}. If not specified, defaults to all options.'
+          optional 'project.id', type: Array, desc: 'Limit search context to list of projects; requires a set of comma-delimited ids like so: {"project.id": ["345e...", "2e45..."]}. Filters out projects to which the user is not allowed access. If not specified, defaults to projects in which the current user has view permission.'
+        end
+        optional :aggs, type: Array, desc: 'List of aggs (aggregates) that should be computed.' do
+          optional :field, type: String, desc: 'The field for which you want to compute agg; valid options are project.name and tags.label. Required if one or more aggs are submitted.'
```

Comment:
I tried this. It makes it difficult to provide the expected error payload for missing parameters, because it halts processing and returns a response before our code gets the chance to rescue from the exception.